packages/transaction-controller/src/utils/validation.ts

```diff
@@ -64,6 +64,23 @@ export function validateTxParams(
   validateGasFeeParams(txParams);
 }
 
+/**
+ * Ensures that error is a nonce issue
+ *
+ * @param error - The error to check
+ * @returns Whether or not the error is a nonce issue
+ */
+// TODO: Replace `any` with type
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isNonceIssue(error: any): boolean {
```

Comment:
Is this technically part of validation as we're not checking any user inputs, rather just testing the type of error.

Would it be clearer inside `TransactionController` where it is used for now?

Reply:
Done in https://github.com/MetaMask/core/pull/3800/commits/5a708d963a81bf9bf4f65d59dc742a56c0a45df8